src/palace/manager/celery/tasks/identifiers.py

```diff
@@ -0,0 +1,159 @@
+from functools import partial
+
+from celery import shared_task
+from celery.canvas import Signature, chord
+from sqlalchemy import select
+from sqlalchemy.orm import raiseload
+
+from palace.manager.celery.task import Task
+from palace.manager.celery.tasks.apply import circulation_apply
+from palace.manager.celery.utils import load_from_id, validate_not_none
+from palace.manager.core.exceptions import PalaceValueError
+from palace.manager.data_layer.circulation import CirculationData
+from palace.manager.service.celery.celery import QueueNames
+from palace.manager.service.redis.models.set import (
+    IdentifierSet,
+    RedisSetKwargs,
+)
+from palace.manager.sqlalchemy.model.collection import Collection
+from palace.manager.sqlalchemy.model.identifier import Identifier
+from palace.manager.sqlalchemy.model.licensing import LicensePool
+
+
+@shared_task(queue=QueueNames.default, bind=True)
+def existing_available_identifiers(task: Task, collection_id: int) -> IdentifierSet:
+    """
+    Retrieves all identifiers that have available licensepools (where licenses_available > 0 and
+    licenses_owned > 0) in the specified collection and returns them as a Redis IdentifierSet.
+
+    This function is designed to be used as part of a chord operation that identifies and
+    marks identifiers not present in a distributor's feed as unavailable.
+
+    See: `create_mark_unavailable_chord`.
+    """
+
+    redis_client = task.services.redis().client()
+    identifier_set = IdentifierSet(redis_client, [task.name, task.request.id])
+
+    try:
+        with task.session() as session:
+            identifiers_query = (
+                select(Identifier)
+                .join(LicensePool)
+                .where(
+                    LicensePool.collection_id == collection_id,
+                    LicensePool.licenses_available != 0,
+                    LicensePool.licenses_owned != 0,
+                )
+                .options(raiseload("*"))
+            )
+
+            for identifiers in (
+                session.execute(identifiers_query).yield_per(100).scalars().partitions()
+            ):
+                identifier_set.add(*identifiers)
+    except:
+        identifier_set.delete()
+        raise
+
+    return identifier_set
+
+
+@shared_task(queue=QueueNames.default, bind=True)
+def mark_identifiers_unavailable(
+    task: Task,
+    identifiers: list[RedisSetKwargs],
```

Comment:
Using "identifiers" here could be a little confusing, especially because we use `identifier` (singular) in this task.

Suggestion: Rename to make it more obvious that these aren't the identifier thingies that we're used to (`Identifier`s or `Identifier.id`s). You'll probably have a better idea of what to name this than I would.

Reply:
I renamed them to `existing_and_active_identifier_sets`